coderd/notifications/notifier.go

```diff
@@ -195,9 +200,18 @@ func (n *notifier) deliver(ctx context.Context, msg database.AcquireNotification
 
 	ctx, cancel := context.WithTimeout(ctx, n.cfg.DispatchTimeout.Value())
 	defer cancel()
-	logger := n.log.With(slog.F("msg_id", msg.ID), slog.F("method", msg.Method))
+	logger := n.log.With(slog.F("msg_id", msg.ID), slog.F("method", msg.Method), slog.F("attempt", msg.AttemptCount+1))
+
+	if msg.AttemptCount > 0 {
+		n.metrics.RetryCount.WithLabelValues(string(n.method), msg.TemplateID.String()).Inc()
+	}
 
+	n.metrics.QueuedSeconds.WithLabelValues(string(n.method)).Observe(msg.QueuedSeconds)
+
+	start := time.Now()
 	retryable, err := deliver(ctx, msg.ID)
+	n.metrics.DispatcherSendSeconds.WithLabelValues(string(n.method)).Observe(time.Since(start).Seconds())
```

Comment:
Reviewer: we may want to distinguish successful vs failed dispatch speed; what do you think?
I left it out because it's probably not very useful in practice.

Reply:
IMHO Dispatching time is fine to be left as is, as a single timer. I would rather consider including the failure reason if possible.